README.md

```diff
@@ -19,12 +19,13 @@ Valid paths are strings conforming to the following BNF syntax.
               <recursive descent> <subpath>
 
 <child> ::= <dot child> | <bracket child>
-<dot child> ::= "." <dotted child name> | ".*"           ; named child (restricted characters) or all children
-<bracket child> ::= "[" <child names> "]"                ; named children
+<dot child> ::= "." <dotted child name> | ".*"          ; named child (restricted characters) or all children
+<bracket child> ::= "[" <child names> "]" | "[" <child names> "]~"                ; named children | property names of children
 <child names> ::= <child name> |
                   <child name> "," <child names> 
 <undotted child> ::= <dotted child name> |               ; named child (restricted characters)
                      <dotted child name><array access> | ; array access of named child
+                     <dotted child name>"~"              ; property name of child
```

Comment:
It would be good to add a test for the case where a child name ends in `~` and make sure the behaviour is as expected. This is tricky because the set of characters allowed in a dotted child name is not documented.

Reply:
this would work the same as if ther was a `.` in the child name, if using the doted child there is a restricted character list but you can use the bracket child syntax for this use case i.e. `$.child['childtest~']~` results in `childtest~`

Reply:
added a test for this as well